MerlinAU.sh

```diff
@@ -2561,6 +2619,348 @@ _GetLatestFWUpdateVersionFromWebsite_()
+            # it does (drive going bad suddenly?) we'll report it here.
+            #-------------------------------------------------------------#
+            Say "Expected directory path $FW_ZIP_BASE_DIR is NOT found."
+            Say "${REDct}**ERROR**${NOct}: Required USB storage device is not connected or not mounted correctly."
+            return 1
+            # Consider how to handle this error. For now, we'll not delete the ZIP file.
+        else
+            keepZIPfile=1
+        fi
+    else
+        #------------------------------------------------------------#
+        # Remove ZIP file here because it may have been corrupted.
+        # Better to download it again and start all over, instead
+        # of trying to figure out why uncompressing it failed.
+        #------------------------------------------------------------#
+        rm -f "$FW_ZIP_FPATH"
+        _SendEMailNotification_ FAILED_FW_UNZIP_STATUS
+        Say "${REDct}**ERROR**${NOct}: Unable to decompress the firmware ZIP file [$FW_ZIP_FPATH]."
+        _return 1
+    fi
+    return 0
+}
+
+##---------------------------------------##
+## Added by ExtremeFiretop [2024-Apr-18] ##
+##---------------------------------------##
+_CopyGnutonFiles_() {
+
+Say "Checking if file management is required"
+
+local copy_success=0
+local copy_attempted=1
+
+# Check and copy the firmware file if different from destination
+if [ "$FW_DL_FPATH" != "${FW_BIN_DIR}/${FW_FileName}.${extension}" ]; then
+    Say "File management is required"
+    copy_attempted=0
+    cp "$FW_DL_FPATH" "$FW_BIN_DIR" && Say "Copying firmware file..." || copy_success=1
+else
+    Say "File management is not required"
+fi
+
+# Check and copy the MD5 file if different from destination
+if [ "$FW_MD5_GITHUB" != "${FW_BIN_DIR}/${FW_FileName}.md5" ]; then
+    copy_attempted=0
+    mv -f "$FW_MD5_GITHUB" "$FW_BIN_DIR" && Say "Moving MD5 file..." || copy_success=1
+fi
+
+# Check and copy the Changelog file if different from destination
+if [ "$FW_Changelog_GITHUB" != "${FW_BIN_DIR}/${FW_FileName}_Changelog.txt" ]; then
+    copy_attempted=0
+    mv -f "$FW_Changelog_GITHUB" "$FW_BIN_DIR" && Say "Moving changelog file..." || copy_success=1
+fi
+
+if [ $copy_attempted -eq 0 ] && [ $copy_success -eq 0 ]
+then
+    #---------------------------------------------------------------#
+    # Check if Gntuon file was downloaded to a USB-attached drive.
+    # Take into account special case for Entware "/opt/" paths.
+    #---------------------------------------------------------------#
+    if ! echo "$FW_DL_FPATH" | grep -qE "^(/tmp/mnt/|/tmp/opt/|/opt/)"
+    then
+        # It's not on a USB drive, so it's safe to delete it
+        rm -f "$FW_DL_FPATH"
+        rm -f "$FW_Changelog_GITHUB"
+        rm -f "$FW_MD5_GITHUB"
+    elif ! _ValidateUSBMountPoint_ "$FW_ZIP_BASE_DIR" 1
+    then
+        -------------------------------------------------------------#
```

Comment:
Error in line **2879**:  missing number char `#` to indicate a comment.

Reply:
Done :)